src/lib/hash.ts

```diff
@@ -192,3 +194,40 @@ class TokenSymbol extends Struct(TokenSymbolPure) {
 function emptyReceiptChainHash() {
   return emptyHashWithPrefix('CodaReceiptEmpty');
 }
+
+function buildSHA(length: 224 | 256 | 384 | 512, nist: boolean) {
+  return {
+    hash(message: (Field | UInt8)[]) {
+      if (message.length === 0) {
```

Comment:
technically `H('')` should work https://en.wikipedia.org/wiki/SHA-3#Examples_of_SHA-3_variants

Reply:
great point!